packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTScrollViewComponentView.mm

```diff
@@ -149,6 +156,104 @@ - (void)dealloc
   [self.scrollViewDelegateSplitter removeAllDelegates];
 }
 
+- (void)_registerKeyboardListener
+{
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(_keyboardWillChangeFrame:)
+                                               name:UIKeyboardWillChangeFrameNotification
+                                             object:nil];
+}
+
+- (void)_unregisterKeyboardListener
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self name:UIKeyboardWillChangeFrameNotification object:nil];
+}
+
+- (void)_keyboardWillChangeFrame:(NSNotification *)notification
+{
+  if (!_automaticallyAdjustKeyboardInsets) {
+    return;
+  }
+  BOOL isHorizontal = _scrollView.contentSize.width > self.frame.size.width;
+  if (isHorizontal) {
+    return;
+  }
```

Comment:
I'm not sure this is a good check.

What if the user is keeping the device in portrait mode but the contentSize has been set to a big picture with width > the frame width, to allow to scroll in 4d?

Reply:
Im also not sure about this check but it was there for the old architecture for a long time... 

There might be a reason for this: 
https://github.com/facebook/react-native/blob/7903dba7ac1f3e686345de8d5df81a238480eef2/packages/react-native/React/Views/ScrollView/RCTScrollView.m#L305